src/heuristic/HeuristicMapper.cpp

```diff
@@ -395,23 +400,46 @@ void HeuristicMapper::mapToMinDistance(const std::uint16_t source,
 }
 
 HeuristicMapper::Node HeuristicMapper::aStarMap(size_t layer) {
-  std::vector<std::uint16_t> consideredQubits{};
-  Node                       node{};
+  std::set<std::uint16_t> consideredQubits{};
```

Comment:
Is it important, that this set is ordered? Or would a `std::unordered_set` do here and be potentially better performing?